core/logger/audit/audit_logger.go

```diff
@@ -0,0 +1,213 @@
+package audit
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/pkg/errors"
+)
+
+const AUDIT_LOGS_CAPACITY = 2048
+
+type AuditLogger interface {
+	Audit(ctx context.Context, eventID EventID, data map[string]interface{})
+}
+
+type AuditLoggerService struct {
+	logger          logger.Logger
+	enabled         bool
+	serviceURL      string
+	serviceHeaders  []serviceHeader
+	jsonWrapperKey  string
+	environmentName string
+	hostname        string
+	localIP         string
+	loggingChannel  chan WrappedAuditLog
+}
+
+// Configurable headers to include in POST to log service
+type serviceHeader struct {
+	header string
+	value  string
+}
+
+type WrappedAuditLog struct {
+	eventID EventID
+	data    map[string]interface{}
```

Comment:
Style nit: If all of the literal `map[string]interface{}{...}` are annoyingly verbose, `any` is interchangeable:
```suggestion
	data    map[string]any
```
Or you could even go a step further by defining a `type audit.Data = map[string]interface{}` alias type? 
```suggestion
	data    Data
```
`audit.Data{"foo": "bar", "key": 42}`